lib/cppcheck.cpp

```diff
@@ -614,6 +618,7 @@ unsigned int CppCheck::checkFile(const std::string& filename, const std::string
                 const std::string &failedToGetAddonInfo = addonInfo.getAddonInfo(addon, mSettings.exename);
                 if (!failedToGetAddonInfo.empty()) {
                     reportOut(failedToGetAddonInfo);
+                    mExitCode = 1;
```

Comment:
@danmar I guess this is not correct. It looks like this variable is meant for analysis errors. But in this case, it is an error in an input file. Is there a better way to set the exit code that also does not need `--error-exitcode=` beeing set?

Reply:
For me this feels ok. It's serious if this does not work.